File: Models/Ticket.js
// models/Ticket.js
const mongoose = require("mongoose");

const ticketSchema = new mongoose.Schema({
  role: {
    type: String,
    required: true,
    enum: ["patient", "doctor", "admin", "superadmin"], // Adjust roles as needed
  },
  issue: {
    type: String,
    required: true,
  },
  status: {
    type: String,
    enum: ["open", "in_progress", "resolved", "closed"],
    default: "open",
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
  review: {
    type: String,
    trim: true,
    default: null, // Optional review field
  },
  user: {
    type: mongoose.Schema.Types.ObjectId,
    refPath: "role",
    required: true,
  },
});

module.exports = mongoose.model("Ticket", ticketSchema);

File: frontend/lib/models/a/ticket.dart
// models/ticket.dart
import 'package:json_annotation/json_annotation.dart';

part 'ticket.g.dart';

@JsonSerializable()
class Ticket {
  final String id;
  final String role;
  final String issue;
  final String status;
  final DateTime createdAt;
  final DateTime? solvedAt;
  final String? review;
  final String user; // Reference to User ID (Patient, Doctor, etc.)
  final List<String> visibleTo;

  Ticket({
    required this.id,
    required this.role,
    required this.issue,
    this.status = 'open',
    required this.createdAt,
    this.solvedAt,
    this.review,
    required this.user,
    this.visibleTo = const ['patient', 'doctor', 'admin', 'superadmin'],
  });

  factory Ticket.fromJson(Map<String, dynamic> json) =>
      _$TicketFromJson(json);

  Map<String, dynamic> toJson() => _$TicketToJson(this);
}
